pkg/security/probe/process_resolver.go

```diff
@@ -240,8 +241,8 @@ func (p *ProcessResolver) enrichEventFromProc(entry *model.ProcessCacheEntry, pr
 	if err != nil {
 		return errors.Wrapf(err, "snapshot failed for %d: couldn't parse kernel capabilities", proc.Pid)
 	}
-	_ = p.resolvers.ResolveProcessContextUser(&entry.ProcessContext)
-	_ = p.resolvers.ResolveProcessContextGroup(&entry.ProcessContext)
+	p.SetProcessUsersGroups(entry)
```

Comment:
Couldn't add a comment on the right line with my IDE, but we should change `len(filledProc.Gids) > 0` to `len(filledProc.Gids) > 4`.